dotcom-rendering/src/web/components/Island.tsx

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+
+import { Placeholder } from './Placeholder';
+
+type Defer = 'idle' | 'visible';
+
+interface HydrateProps {
+	defer?: Defer;
+	ssr?: true;
+	placeholderHeight?: undefined;
+	children: JSX.Element;
+}
+
+interface PortalProps {
```

Comment:
If we're moving away from the `Portal` nomenclature, should we try something like `ClientOnlyProps`?

Reply:
Agreed, done